app/Services/Model/Imp/ScheduleService.php

```diff
@@ -0,0 +1,262 @@
+
+            $schedule->removeScheduledSummitEvent($schedule_event);
+
+            return $schedule_event;
+        });
+    }
+
+    /**
+     *@inheritDoc
+     */
+    public function publishAll(string $source, int $summit_id, array $payload):SummitProposedSchedule
+    {
+        return $this->tx_service->transaction(function () use ($source, $summit_id, $payload) {
+
+            $schedules = $this->schedule_repository->getBySourceAndSummitId($source, $summit_id);
+
+            if (count($schedules) == 0) {
+                throw new EntityNotFoundException("schedule with source {$source} does not exists!");
+            }
+
+            $schedule = $schedules[0];
+
+            $filtered_schedule_events = [];
+
+            if (isset($payload['presentation_ids'])) {
+                //filter criteria to promote schedule events by event_ids
+                $event_ids = $payload['presentation_ids'];
+                foreach ($event_ids as $event_id) {
+                    $event = $this->event_repository->getById(intval($event_id));
+                    if (!$event instanceof SummitEvent) continue;
+                    $schedule_event = $schedule->getScheduledSummitEventByEvent($event);
+                    if (is_null($schedule_event)) continue;
+                    $filtered_schedule_events[] = $schedule_event;
+                }
+            } else {
+                //filter criteria to promote schedule events by start_date, end_date and location_id
+                $summit = $schedule->getSummit();
+                $start_date = null;
+                $end_date = null;
+                $location = null;
+                if (isset($payload['start_date'])) {
+                    $start_date = $summit->parseDateTime(intval($payload['start_date']));
+                }
+                if (isset($payload['end_date'])) {
+                    $end_date = $summit->parseDateTime(intval($payload['end_date']));
+                }
+                if (isset($payload['location_id'])) {
+                    $location = $summit->getLocation(intval($payload['location_id']));
+                }
+                $filtered_schedule_events =
+                    $schedule->getScheduledSummitEventsByLocationAndDateRange($start_date, $end_date, $location);
+            }
+
+            if (count($filtered_schedule_events) > 0) {
+                $schedule->clearScheduledSummitEvents();
+
+                foreach ($filtered_schedule_events as $filtered_schedule_event) {
+                    $event = $filtered_schedule_event->getSummitEvent();
+                    if ($event instanceof SummitEvent) {
+                        $event->setStartDate($filtered_schedule_event->getStartDate());
+                        $event->setEndDate($filtered_schedule_event->getEndDate());
+                        $event->setLocation($filtered_schedule_event->getLocation());
```

Comment:
this publishing process is not correct
u are only setting start/end date and location
please check SummitService::publishEvent, i do think that logic should be moved up on the hierarchy (AbstractPublishService) and both service should be used

Reply:
done